tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/DataType.java

```diff
@@ -35,13 +34,13 @@
   public interface TensorMapper<T> {
 
     /**
-     * Maps tensor memory to a data structure for manipulating elements of this type.
+     * Maps the tensor memory to a n-dimensional typed data space.
      *
-     * @param nativeTensor pointer to the native tensor
-     * @param shape the shape of the tensor
-     * @return data structure of elements of this type
+     * @param tensor the tensor to map in its raw nature
+     * @param nativeHandle native handle of the tensor
+     * @return a typed tensor of type {@code T}
      */
-    T apply(TF_Tensor nativeTensor, Shape shape);
+    T apply(RawTensor tensor, TF_Tensor nativeHandle);
```

Comment:
The reason I'm passing `nativeHandle` as a separate parameter here is because I want to keep `RawTensor.nativeHandle()` package-private, reserving its use to what is located under `org.tensorflow` only. But as implementors of `TensorMapper` are outside this package, `apply` serves also as a visitor to `RawTensor` internal representation.

Reply:
Convincing! Perhaps incorporate this explanation as a comment?

Reply:
> We could add a default method that accepts a RawTensor and unpacks it into the RawTensor and the TF_Tensor and calls apply

Current signature for `apply` also ensure that only classes of the `org.tensorflow` package can invoke it, since you shouldn't be able to have a reference to a tensor native handle. Just accepting `RawTensor` will open it up to the outside world. Now, let's not be zealous about securing the tensor native handle neither, I think this is a good trade off, I'll make the changes.

Reply:
Sure, that's a good point. If it's something that's designed to be locked off when we eventually have modules then maybe we should just document it that way and move on.